pkg/update/update.go

```diff
@@ -485,26 +483,21 @@ func updateDnsPod(httpClient *http.Client, domain, host, token, ip string) (newI
 		return "", err
 	}
 	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
-	resp, err := httpClient.Do(req)
+	var content []byte
+	_, content, err = network.DoHTTPRequest(httpClient, req)
 	if err != nil {
 		return "", err
 	}
-	respBody, err := ioutil.ReadAll(resp.Body)
-	resp.Body.Close()
-	if err != nil {
-		return "", err
-	}
-	type RecordResp struct {
+	recordResp := new(struct {
```

Comment:
just do 

```go
var recordResp struct {
    Records []struct {
        ID    string `json:"id"`
        Value string `json:"value"`
        Type  string `json:"type"`
        Name  string `json:"name"`
        Line  string `json:"line"`
    } `json:"records"`
}
```

The `new` keyboard is cumbersome and there is no need for pointers here `*`.

Adapt also the next line to 

```go
err = json.Unmarshal(content, &recordResp)
```